fix for GHSA-gj48-438w-jh9v (clean() does not sanitize dangerous URI schemes in formaction attributes)

see https://github.com/mozilla/bleach/commit/970df58e9f

author: Rohit Goswami (HaoZeke)

diff --git a/bleach/_vendor/03_html5lib_formaction.patch b/bleach/_vendor/03_html5lib_formaction.patch
new file mode 100644
index 0000000..0b8b050
--- /dev/null
+++ b/bleach/_vendor/03_html5lib_formaction.patch
@@ -0,0 +1,12 @@
+diff --git bleach/_vendor/html5lib/filters/sanitizer.py b/bleach/_vendor/html5lib/filters/sanitizer.py
+index a37c9d8..dd5a6a6 100644
+--- bleach/_vendor/html5lib/filters/sanitizer.py
++++ bleach/_vendor/html5lib/filters/sanitizer.py
+@@ -534,6 +534,7 @@ attr_val_is_uri = frozenset((
+     (None, 'dynsrc'),
+     (None, 'lowsrc'),
+     (None, 'ping'),
++    (None, 'formaction'),
+     (namespaces['xlink'], 'href'),
+     (namespaces['xml'], 'base'),
+ ))
diff --git a/bleach/_vendor/html5lib/filters/sanitizer.py b/bleach/_vendor/html5lib/filters/sanitizer.py
index a37c9d8..dd5a6a6 100644
--- a/bleach/_vendor/html5lib/filters/sanitizer.py
+++ b/bleach/_vendor/html5lib/filters/sanitizer.py
@@ -534,6 +534,7 @@ attr_val_is_uri = frozenset((
     (None, 'dynsrc'),
     (None, 'lowsrc'),
     (None, 'ping'),
+    (None, 'formaction'),
     (namespaces['xlink'], 'href'),
     (namespaces['xml'], 'base'),
 ))
diff --git a/bleach/_vendor/vendor_install.sh b/bleach/_vendor/vendor_install.sh
index df88834..875bb78 100755
--- a/bleach/_vendor/vendor_install.sh
+++ b/bleach/_vendor/vendor_install.sh
@@ -13,6 +13,7 @@ pip install --no-binary all --no-compile --no-deps -r "${BLEACH_VENDOR_DIR}/vend
 # Apply patches
 (cd "${DEST}" && patch -p2 < 01_html5lib_six.patch)
 (cd "${DEST}" && patch -p2 < 02_html5lib_wbr.patch)
+(cd "${DEST}" && patch -p2 < 03_html5lib_formaction.patch)
 
 # install Python 3.6.14 urllib.urlparse for #536
 curl --proto '=https' --tlsv1.2 -o "${DEST}/parse.py" https://raw.githubusercontent.com/python/cpython/v3.6.14/Lib/urllib/parse.py
